tests/Invoke-DbaDiagnosticQuery.Tests.ps1

```diff
@@ -1,16 +1,98 @@
+    }
+
+
+    Context "verifying output when running queries" {
         It "runs a specific query" {
-            $results = Invoke-DbaDiagnosticQuery -SqlInstance $script:instance2 -QueryName 'Memory Clerk Usage' *>&1
-            $results.Name.Count | Should Be 1
+            $results = Invoke-DbaDiagnosticQuery -SqlInstance $script:instance2 -QueryName 'Memory Clerk Usage'
+            @($results).Count | Should -Be 1
         }
         It "works with DatabaseSpecific" {
-            $results = Invoke-DbaDiagnosticQuery -SqlInstance $script:instance2 -DatabaseSpecific *>&1
-            $results.Name.Count -gt 10 | Should Be $true
+            $results = Invoke-DbaDiagnosticQuery -SqlInstance $script:instance2 -DatabaseSpecific
+            @($results).Count | Should -BeGreaterThan 10
+        }
+        It "works with Exclude Databases provided" {
+            $results = Invoke-DbaDiagnosticQuery -SqlInstance $script:instance2 -DatabaseSpecific -ExcludeDatabase $database2
+            @($results | Where-Object {$_.DatabaseName -eq $Database1}).Count | Should -BeGreaterThan 1
+            @($results | Where-Object {$_.DatabaseName -eq $Database2}).Count | Should -Be 0
+        }
+    }
+
+    context "verifying output when exporting queries as files instead of running" {
+
+        It "exports queries to sql files without running" {
+            Invoke-DbaDiagnosticQuery -SqlInstance $script:instance2 -ExportQueries -QueryName 'Memory Clerk Usage' -OutputPath $script:PesterOutputPath
+            @(Get-ChildItem -path $script:PesterOutputPath -filter *.sql).Count | Should -Be 1
+        }
+
+        It "verifies returns object with required properties when running with whatif" {
+            [System.Management.Automation.PSObject[]]$results = Invoke-DbaDiagnosticQuery -SqlInstance $script:instance2 -WhatIf
+
+            @($results).count                                                             | Should -BeGreaterThan 10
+            (($results).ComputerName     | Where-Object {$_ -eq $null}).count             | Should -Be 0
+            (($results).InstanceName     | Where-Object {$_ -eq $null}).count             | Should -Be 0
+            (($results).SqlInstance      | Where-Object {$_ -eq $null}).count             | Should -Be 0
+            (($results).Number           | Where-Object {$_ -eq $null}).count             | Should -Be 0
+            (($results).Name             | Where-Object {$_ -eq $null}).count             | Should -Be 0
+            (($results).Description      | Where-Object {$_ -eq $null}).count             | Should -Be 0
+            (($results).DatabaseSpecific | Where-Object {$_ -eq $null}).count             | Should -Be 0
+            (($results).DatabaseName).count                                               | Should -Be @($results).count
+            (($results).Notes).count                                                      | Should -Be @($results).count
```

Comment:
Consistency! Either `count` or `Count` (preferred the latter) but definitely not both